src/dynsight/_internal/trajectory/trajectory.py

```diff
@@ -61,131 +84,206 @@ def get_onion(self, delta_t: int) -> OnionInsight:
-        if self.dataset.ndim == UNIVAR_DIM:
+    def plot_one_trj(
+        self,
+        file_path: Path,
+        data_insight: Insight,
+        particle_id: int,
+    ) -> None:
+        """Plot one particle's trajectory colored according to clustering."""
+        if data_insight.dataset.ndim == UNIVAR_DIM:
             dynsight.onion.plot.plot_one_trj_uni(
-                file_name,
+                file_path,
                 particle_id,
                 self.reshaped_data,
-                self.dataset.shape[0],
+                data_insight.dataset.shape[0],
                 self.labels,
             )
         else:
             dynsight.onion.plot.plot_one_trj_multi(
-                file_name,
+                file_path,
                 particle_id,
-                self.delta_t,
-                self.dataset,
+                self.meta["delta_t"],
+                data_insight.dataset,
                 self.labels,
             )
 
-    def plot_medoids(self, file_name: str) -> None:
-        if self.dataset.ndim == UNIVAR_DIM:
+    def plot_medoids(self, file_path: Path, data_insight: Insight) -> None:
+        """Plot the average sequence of each onion cluster."""
+        if data_insight.dataset.ndim == UNIVAR_DIM:
             dynsight.onion.plot.plot_medoids_uni(
-                file_name,
+                file_path,
                 self.reshaped_data,
                 self.labels,
             )
         else:
             dynsight.onion.plot.plot_medoids_multi(
-                file_name,
-                self.delta_t,
-                self.dataset,
+                file_path,
+                self.meta["delta_t"],
+                data_insight.dataset,
                 self.labels,
             )
 
-    def plot_state_populations(self, file_name: str) -> None:
+    def plot_state_populations(
+        self,
+        file_path: Path,
+        data_insight: Insight,
+    ) -> None:
+        """Plot each state's population along the trajectory."""
         dynsight.onion.plot.plot_state_populations(
-            file_name,
-            self.dataset.shape[0],
-            self.delta_t,
+            file_path,
+            data_insight.dataset.shape[0],
+            self.meta["delta_t"],
             self.labels,
         )
 
-    def plot_sankey(self, file_name: str, frame_list: list[int]) -> None:
+    def plot_sankey(
+        self,
+        file_path: Path,
+        data_insight: Insight,
+        frame_list: list[int],
+    ) -> None:
+        """Plot the Sankey diagram of the onion clustering."""
         dynsight.onion.plot.plot_sankey(
-            file_name,
+            file_path,
             self.labels,
-            self.dataset.shape[0],
+            data_insight.dataset.shape[0],
             frame_list,
         )
 
 
-@dataclass
+@dataclass(frozen=True)
 class Trj:
-    """Contains a many-body trajectory."""
+    """Contains a trajectory."""
+
+    universe: MDAnalysis.Universe = field()
```

Comment:
I think this looks good, Now we start thinking about testing.

- [ ] I think we need tests that cover all init types of Trj. 
- [ ] Maybe we rewrite all existing tests to use the Trj as well?

Reply:
I am thinking about future changes - calling this a `universe` is assuming it is equivalent to an mdanalysis universe. Maybe state in documentation a `warning` that this is under development and may change